packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -1055,7 +1161,64 @@ export function attach(
     pendingOperations.push(op);
   }
 
+  function reevaluateErrorsAndWarnings() {
+    fibersWithChangedErrorOrWarningCounts.clear();
+    fiberToErrorsMap.forEach((countMap, fiberID) => {
+      fibersWithChangedErrorOrWarningCounts.add(fiberID);
+    });
+    fiberToWarningsMap.forEach((countMap, fiberID) => {
+      fibersWithChangedErrorOrWarningCounts.add(fiberID);
+    });
+    recordPendingErrorsAndWarnings();
+  }
+
+  function recordPendingErrorsAndWarnings() {
+    fibersWithChangedErrorOrWarningCounts.forEach(fiberID => {
+      const fiber = idToFiberMap.get(fiberID);
+      if (fiber != null) {
+        // Don't send updates for Fibers that didn't mount due to e.g. Suspense or an error boundary.
+        // We may also need to clean up after ourselves to avoid leaks.
+        // See inline comments in onErrorOrWarning() for more info.
+        if (isFiberMountedImpl(fiber) !== MOUNTED) {
+          fiberToIDMap.delete(fiber);
+          idToFiberMap.delete(fiberID);
+          primaryFibers.delete(fiber);
+          return;
+        }
```

Comment:
This is the only part of the code that I don't feel _great_ about. I'm not sure of a better way to handle it at the moment though.